src/simularium/VisDataCache.ts

```diff
@@ -78,18 +67,17 @@ class VisDataCache {
      * returns the node if found, otherwise returns null,
      * starts at head if firstNode is not provided.
      */
-    private walkLinkedList(
+    private findInLinkedList(
         condition: (data: CacheNode) => boolean,
         firstNode?: CacheNode
-    ): CacheNode | null {
+    ): CacheNode | undefined {
         let currentNode = firstNode || this.head;
         while (currentNode) {
             if (condition(currentNode)) {
                 return currentNode;
             }
             currentNode = currentNode.next;
         }
-        return null;
```

Comment:
miiiight still keep the explicit `return undefined;` for completeness, even though `undefined` is implicitly returned anyways